### PR TITLE
Logs: Add CSV Logger, modify logging from 4 GPUs to logging all-reduc…

### PR DESCRIPTION
**Pull request by vietngth**
1. Modify to log the aggregated values only (all_reduced value in 4 GPU each iteration/epoch) - the old version logs each GPU separately
2. Remove the validation_epoch_end() function because they are redundant, Trainer logger can already take care of the average value for us
3. Add CSVLogger along with TF logs just for debugging
4. Add console logging after aggregating the results
5. Add the JSON file at the end to store some static values like trainable_params, learning rate, weight decay, etc.

Please check if I have handled the result_path correctly to store the resulted architectures

### CODE REVIEW

**Review by emreds — APPROVED** [review]
We also made some changes for the pipeline in this PR.

**Comment by emreds** [comment]
Looks great! Thank you @vietngth
